### PR TITLE
feat(`ListOnOpenseaProposal`): support listing on OpenSea as dutch auction

### PR DESCRIPTION
**Pull request by 0xble**

### CODE REVIEW

File: contracts/proposals/ListOnOpenseaProposal.sol [hunk 1]
```diff
@@ -68,10 +72,17 @@ abstract contract ListOnOpenseaProposal is ZoraHelpers {
...
+        uint256 endPrice,
         uint256 expiry
     );
-    event OpenseaOrderSold(bytes32 orderHash, IERC721 token, uint256 tokenId, uint256 listPrice);
+    event OpenseaOrderSold(
+        bytes32 orderHash,
+        IERC721 token,
+        uint256 tokenId,
+        uint256 startPrice,
+        uint256 endPrice
```
Comment (merklejerk) [inline contracts/proposals/ListOnOpenseaProposal.sol:84]: wouldn't it be more useful to emit the price it sold at

File: contracts/proposals/ListOnOpenseaProposal.sol [hunk 2]
```diff
@@ -255,7 +270,14 @@ abstract contract ListOnOpenseaProposal is ZoraHelpers {
...
         orderParams.salt = uint256(bytes32(domainHashPrefix));
-        orderParams.conduitKey = conduitKey;
+        {
+            // Approve OpenSea's conduit to spend our NFT. This should revert if we
+            // do not own the NFT.
+            bytes32 conduitKey = _GLOBALS.getBytes32(LibGlobals.GLOBAL_OPENSEA_CONDUIT_KEY);
+            (address conduit, ) = CONDUIT_CONTROLLER.getConduit(conduitKey);
+            token.approve(conduit, tokenId);
+            orderParams.conduitKey = conduitKey;
+        }
```
Comment (merklejerk) [inline contracts/proposals/ListOnOpenseaProposal.sol:280]: awkward placement for a side-effect imo, maybe move to L315 and you can just work on `orderParams.conduitKey` directly?

**Review by merklejerk — APPROVED** [review]
minor suggestions but looks good otherwise

**Comment by johncpalmer** [comment]
wow

**Pull request merged**